### ISSUE TITLE
TypeError on MacOS window close

### ISSUE THREAD

**Issue by Abeardevil** [pythonarcade/arcade#382]
## Bug Report
This was actually discovered during your 2019 PyCon tutorial. This only seems to affect MacOS devices.

### Actual behavior:
Error thrown when closing window:

```
Written to (null)
Traceback (most recent call last):
  File "/Users/ccameron/workspace/pycon/2d_platformer/.venv/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 755, in __call__
    result = f(objc_id, self.selector, *args)
ctypes.ArgumentError: argument 1: <class 'RecursionError'>: maximum recursion depth exceeded in __instancecheck__

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 232, in 'calling callback function'
  File "/Users/ccameron/workspace/pycon/2d_platformer/.venv/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/ccameron/workspace/pycon/2d_platformer/.venv/lib/python3.7/site-packages/pyglet/window/cocoa/pyglet_window.py", line 59, in nextEventMatchingMask_untilDate_inMode_dequeue_
    if self.inLiveResize():
  File "/Users/ccameron/workspace/pycon/2d_platformer/.venv/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 785, in __call__
    return self.method(self.objc_id, *args)
  File "/Users/ccameron/workspace/pycon/2d_platformer/.venv/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 764, in __call__
    error.args += ('selector = ' + self.name,
TypeError: can only concatenate str (not "bytes") to str
```

### Expected behavior:
No error when closing window.
### Steps to reproduce/example code:
Simple example code:
```
import arcade

SCREEN_WIDTH = 1000
SCREEN_HEIGHT = 650
SCREEN_TITLE = "Caleb's Platformer!"

class MyGame(arcade.Window):
    def __init__(self):
        super().__init__(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE)
        arcade.set_background_color(arcade.csscolor.CORNFLOWER_BLUE)
    
    def setup(self):
        pass
    
    def on_draw(self):
        arcade.start_render()

def main():
    window = MyGame()
    window.setup()
    arcade.run()

if __name__ == '__main__':
    main()
```

**Comment by Abeardevil** [comment]
I'm on MacOS 10.14.3

**Comment by Abeardevil** [comment]
This also causes python to exit with error code 134.

```
(.venv) 2d_platformer/platform_tutorial » echo $?
134
```

**Comment by pvcraven** [comment]
Closing. Haven't been able to reproduce. Not sure if this is still happening with new versions of pyglet.

**Comment by donpatrice** [comment]
I can reproduce that with pygelt 1.4.10 on a MacBook Pro (13-inch, 2018, Four Thunderbolt 3 Ports) with macos 10.15.2. So this issue should be reopened.

**Comment by pvcraven** [comment]
Any chance you could please post the full stack trace?

**Comment by donpatrice** [comment]
Sure!

```
2020-01-30 15:26:46.227 Python[87099:3928223] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to (null)
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 237, in 'calling callback function'
  File "/Users/patrik/.pyenv/versions/3.8.1/envs/pyglet_test/lib/python3.8/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/patrik/.pyenv/versions/3.8.1/envs/pyglet_test/lib/python3.8/site-packages/pyglet/window/cocoa/pyglet_window.py", line 65, in nextEventMatchingMask_untilDate_inMode_dequeue_
    if self.inLiveResize():
  File "/Users/patrik/.pyenv/versions/3.8.1/envs/pyglet_test/lib/python3.8/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 785, in __call__
    return self.method(self.objc_id, *args)
  File "/Users/patrik/.pyenv/versions/3.8.1/envs/pyglet_test/lib/python3.8/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 755, in __call__
    result = f(objc_id, self.selector, *args)
ctypes.ArgumentError: ("argument 1: <class 'RecursionError'>: maximum recursion depth exceeded in __instancecheck__", "selector = b'inLiveResize'", "argtypes =[<class 'ctypes.c_void_p'>, <class 'ctypes.c_void_p'>]", "encoding = b'c16@0:8'")
[1]    87099 segmentation fault  python example.py
```

**Comment by marty331** [comment]
Here's my ST, I'm on MacOS 10.15.4.
`2020-04-01 22:33:42.177 Python[62932:2160162] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to (null)
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 232, in 'calling callback function'
  File "/Users/marty331/Python/template-master/env/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/marty331/Python/template-master/env/lib/python3.7/site-packages/pyglet/window/cocoa/pyglet_window.py", line 64, in nextEventMatchingMask_untilDate_inMode_dequeue_
    if self.inLiveResize():
  File "/Users/marty331/Python/template-master/env/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 785, in __call__
    return self.method(self.objc_id, *args)
  File "/Users/marty331/Python/template-master/env/lib/python3.7/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 755, in __call__
    result = f(objc_id, self.selector, *args)
ctypes.ArgumentError: ("argument 1: <class 'RecursionError'>: maximum recursion depth exceeded while calling a Python object", "selector = b'inLiveResize'", "argtypes =[<class 'ctypes.c_void_p'>, <class 'ctypes.c_void_p'>]", "encoding = b'c16@0:8'")
objc[62932]: Attempt to use unknown class 0x7ffee950c210.

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)`

**Comment by pvcraven** [comment]
Man I wish I had a Mac that I could reproduce this on and debug it.

**Comment by pushfoo** [comment]
As of writing, I can't replicate this issue. 
I'm on OS X 10.12 and my pyglet version is 1.5.2.

**Comment by pvcraven** [comment]
It only seems to happen on some macs. I'm not sure what the common factor is.

**Comment by pushfoo** [comment]
I have a suspicion it only happens starting on some version after 10.12.

**Comment by einarf** [comment]
I think it's only a problem on macs from 2018 and later. I know the pyglet guys are aware of it and it's on the list of things to solve. I don't see this issue on my 2013 macbook. Otherwise I would try to solve it 😄

**Comment by marty331** [comment]
@einarf I have a 2017 mac that it happens on.

**Comment by pushfoo** [comment]
@marty331 what's your OS X version? Also, do you have a discrete gpu or an integrated intel one?

**Comment by marty331** [comment]
@pushfoo I'm running 10.15.4 and have Intel HD Graphics 630 1536 MB

**Comment by RyanSchaefer** [comment]
@pvcraven Connecting you with this, they had a reproducible way to generate the issue: https://github.com/pyglet/pyglet/issues/5

https://github.com/olehermanse/cocoa-python/commit/dc536717f0dcf9f008108a882ccb0955135160eb

**Comment by einarf** [comment]
The pyglet people are still trying to find a solution to it. We're in their discord server and speak regularly.

**Comment by mstfymrtc** [comment]
This also same for me. When i run arcade program, it stucks and when i try to close it, it behaves like it's in the endless loop and insist on not to close. At last, i just do force quit. MacOS Catalina 10.15.4

**Comment by einarf** [comment]
Having some kind of exit key that calls `window.close()` is probably the best workaround currently.

**Comment by einarf** [comment]
Should be solved if pyglet 1.5.6
